GRDB/Core/Database.swift

```diff
@@ -1128,3 +1165,28 @@ extension Database {
         }
     }
 }
+
+#if SQLITE_HAS_CODEC && GRDB_SQLITE_SEE
+extension Database {
+    /// An SQLite Encryption Extension encryption type
+    ///
+    /// The raw value is the string to prefix a key with
+    public enum EncryptionType: String {
```

Comment:
The fourth algorithm is AES128 but implemented using a different library.

You can also choose to only compile a single algorithm into the library. In this case the key prefix is considered part of the key. This is one of the parts where I'm unsure about the design. I haven't made the encryption algorithm optional as it *possibly* should be. Since the recommended method is to include all three, I believe it is more convenient for the majority to take this approach. If anyone complains, it can always be changed later.

Reply:
All right, @benrb, this sounds OK.